tsl/src/fdw/scan_plan.c

```diff
@@ -869,6 +869,64 @@ foreign_grouping_ok(PlannerInfo *root, RelOptInfo *grouped_rel, GroupPathExtraDa
 	return true;
 }
 
+/*
+ * Find an equivalence class member expression to be computed as a sort column
+ * in the given target.
+ */
+static Expr *
+find_em_expr_for_input_target(PlannerInfo *root, EquivalenceClass *ec, PathTarget *target)
+{
+	ListCell *lc1;
+	int i;
+
+	i = 0;
+	foreach (lc1, target->exprs)
+	{
+		Expr *expr = (Expr *) lfirst(lc1);
+		Index sgref = get_pathtarget_sortgroupref(target, i);
+		ListCell *lc2;
+
+		/* Ignore non-sort expressions */
+		if (sgref == 0 || get_sortgroupref_clause_noerr(sgref, root->parse->sortClause) == NULL)
+		{
+			i++;
+			continue;
+		}
+
+		/* We ignore binary-compatible relabeling on both ends */
+		while (expr && IsA(expr, RelabelType))
+			expr = ((RelabelType *) expr)->arg;
+
+		/* Locate an EquivalenceClass member matching this expr, if any */
+		foreach (lc2, ec->ec_members)
+		{
+			EquivalenceMember *em = (EquivalenceMember *) lfirst(lc2);
+			Expr *em_expr;
+
+			/* Don't match constants */
+			if (em->em_is_const)
+				continue;
+
+			/* Ignore child members */
+			if (em->em_is_child)
+				continue;
+
+			/* Match if same expression (after stripping relabel) */
+			em_expr = em->em_expr;
+			while (em_expr && IsA(em_expr, RelabelType))
+				em_expr = ((RelabelType *) em_expr)->arg;
+
+			if (equal(em_expr, expr))
+				return em->em_expr;
```

Comment:
It looks like the code always exits here on first iteration with current tests, as line 923 is never executed.